src/app/vault/password-generator.component.html

```diff
@@ -62,148 +90,356 @@
+                    type="checkbox"
+                    (change)="savePasswordOptions()"
+                    [disabled]="enforcedPasswordPolicyOptions?.useNumbers"
+                    [(ngModel)]="passwordOptions.number"
+                  />
+                </div>
+                <div class="box-content-row box-content-row-checkbox" appBoxRow>
+                  <label for="special">!@#$%^&*</label>
+                  <input
+                    id="special"
+                    type="checkbox"
+                    (change)="savePasswordOptions()"
+                    [disabled]="enforcedPasswordPolicyOptions?.useSpecial"
+                    [(ngModel)]="passwordOptions.special"
+                  />
+                </div>
+              </div>
+            </div>
+            <div class="box" [hidden]="!showOptions">
+              <div class="box-content condensed">
+                <div class="box-content-row box-content-row-input" appBoxRow>
+                  <label for="min-number">{{ "minNumbers" | i18n }}</label>
+                  <input
+                    id="min-number"
+                    type="number"
+                    min="0"
+                    max="9"
+                    (blur)="savePasswordOptions()"
+                    [(ngModel)]="passwordOptions.minNumber"
+                  />
+                </div>
+                <div class="box-content-row box-content-row-input" appBoxRow>
+                  <label for="min-special">{{ "minSpecial" | i18n }}</label>
+                  <input
+                    id="min-special"
+                    type="number"
+                    min="0"
+                    max="9"
+                    (blur)="savePasswordOptions()"
+                    [(ngModel)]="passwordOptions.minSpecial"
+                  />
+                </div>
+                <div class="box-content-row box-content-row-checkbox" appBoxRow>
+                  <label for="ambiguous">{{ "ambiguous" | i18n }}</label>
+                  <input
+                    id="ambiguous"
+                    type="checkbox"
+                    (change)="savePasswordOptions()"
+                    [(ngModel)]="avoidAmbiguous"
+                  />
+                </div>
+              </div>
+            </div>
+          </ng-container>
+        </ng-container>
+        <ng-container *ngIf="type === 'username'">
+          <div class="box">
+            <div class="box-header">
+              <button
+                type="button"
+                (click)="toggleOptions()"
+                appA11yTitle="{{ 'toggleVisibility' | i18n }}"
+              >
+                <i class="bwi bwi-plus-square" aria-hidden="true" [hidden]="showOptions"></i>
+                <i class="bwi bwi-minus-square" aria-hidden="true" [hidden]="!showOptions"></i>
+                {{ "options" | i18n }}
+              </button>
+            </div>
+            <div class="box-content condensed" [hidden]="!showOptions">
```

Comment:
Any reason to use `[hidden]` instead of `*ngIf` here (and below)?

Reply:
Not sure I see a reason. Was just following existing code for password generator.

Reply:
Fair enough. I don't think it'll make a difference in this case, so we can leave it.